### PR TITLE
Update CMK Unencrypted Storage query for AWS CloudFormation 

### LINKED ISSUE #2149: Update CMK Unencrypted Storage query for AWS CloudFormation
### Platform
AWS CloudFormation

### Provider
AWS

### Description
Updating the query to reduce the number of false positives. Changing the way to determine if a resource can host a database to a list, instead of using keywords.

### PR DESCRIPTION
**Pull request by Ruben-Silva**
Closes #2149

**Proposed Changes**

- Changing the way to determine if a resource can host a database to a list, instead of using keywords;
- Added new samples;
- Deepened a searchKey.

I submit this contribution under Apache-2.0 license.

### CODE REVIEW

File: assets/queries/cloudFormation/cmk_unencrypted_storage/query.rego [hunk 1]
```diff
@@ -49,48 +49,43 @@ CxPolicy[result] {
...
+		"keyActualValue": sprintf("Resources.%s.Properties.Encrypted is false", [key]),
 	}
 }
 
-isDBName(name) = result {
-	possibleDBs := {"couchbase", "riak", "redis", "hbase", "Oracle", "SAP Hana", "Postgres", "cassandra", "hadoop", "Mongo", "Neo4j", "DB", "SQL"}
-	result := contains(name, possibleDBs[_])
-} else {
-	result := false
+checkTypeS(type){
```
Comment (ruigomescx) [inline assets/queries/cloudFormation/cmk_unencrypted_storage/query.rego:83]: why have this checkTypeS and checkTypeK?

Comment (Ruben-Silva) [reply assets/queries/cloudFormation/cmk_unencrypted_storage/query.rego:83]: To check for the different attributes, `checkTypeS` is for the types in which the existence of "StorageEncrypted" is to be verified and `checkTypeK` is for when "KmsKeyId" is to be verified.

Comment (ruigomescx) [reply assets/queries/cloudFormation/cmk_unencrypted_storage/query.rego:83]: okay, then rename the function to have a more explicit name, checkTypeStorage

Comment (Ruben-Silva) [reply assets/queries/cloudFormation/cmk_unencrypted_storage/query.rego]: Done

**Review by ruigomescx — APPROVED** [review]

**Pull request merged**